src/test/java/uk/ac/ebi/atlas/search/CellTypeSearchByOrganismPartDaoIT.java

```diff
@@ -0,0 +1,74 @@
+package uk.ac.ebi.atlas.search;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+import uk.ac.ebi.atlas.configuration.TestConfig;
+import uk.ac.ebi.atlas.solr.cloud.SolrCloudCollectionProxyFactory;
+
+import javax.inject.Inject;
+import javax.sql.DataSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@WebAppConfiguration
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = TestConfig.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CellTypeSearchByOrganismPartDaoIT {
+    @Inject
+    private DataSource dataSource;
+
+    @Inject
+    private SolrCloudCollectionProxyFactory collectionProxyFactory;
+
+    private CellTypeSearchByOrganismPartDao subject;
+
+    @BeforeAll
+    void populateDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(new ClassPathResource("fixtures/experiment-fixture.sql"));
+        populator.execute(dataSource);
+    }
+
+    @AfterAll
+    void cleanDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(new ClassPathResource("fixtures/experiment-delete.sql"));
+        populator.execute(dataSource);
+    }
+
+    @BeforeEach
+    void setUp() {
+        subject = new CellTypeSearchByOrganismPartDao(collectionProxyFactory);
+    }
+
+    @Test
+    void invalidCellTypeMetadataSearch() {
+        assertThat(subject.getCellTypeMetadata("E-MTAB-5061", "foobar"))
+                .isEmpty();
+    }
+
+    @Test
+    void validCellTypeMetadataSearch() {
+        // pancreas
+        var cellTypesInPancreas = subject.getCellTypeMetadata("E-MTAB-5061", "http://purl.obolibrary.org/obo/UBERON_0001264");
+        assertThat(cellTypesInPancreas)
+                .isNotEmpty();
+        // islet of Langerhans
+        var cellTypesInIsletOfLangerhans = subject.getCellTypeMetadata("E-MTAB-5061", "http://purl.obolibrary.org/obo/UBERON_0000006");
+        assertThat(cellTypesInIsletOfLangerhans)
+                .isNotEmpty();
+
+        assertThat(cellTypesInPancreas)
+                .containsAll(cellTypesInIsletOfLangerhans);
+    }
+}
```

Comment:
I will also add one more test case in the scenario where we search for two `organsOrOrganismPart` which are not related to each other and should not contain each other `cell types`. For example, searching for `skin` and `kidney` should not have common cell types. I hope it makes sense.

Reply:
Have we got any experiment with multiple organs in our test datasets? I don’t feel like loading Tabula Muris in our laptops. :D

Reply:
![1_ZyE6yAwn5DlOfsO6UTC8Yw](https://user-images.githubusercontent.com/13108541/94469131-7945db00-01bd-11eb-8e3e-dab39f68b48c.jpeg)

Reply:
So I played a bit with the test experiment @upendrakumbham mentioned, E-EHCA-2, and it turns out that the test case mentioned above doesn’t hold.

For example, the following query shows the `inferred cell type - authors labels` from a set of skin cells:
```solr
cell_id:(21784_6_131 OR 21784_6_137 OR 21784_6_169 OR 21784_6_181 OR 21784_6_183 OR 21784_6_191 OR 21784_6_225 OR 21784_6_227 OR 21784_6_231 OR 21784_6_267 OR 21784_6_279 OR 21784_6_321 OR 21784_6_329 OR 21784_6_33 OR 22467_8_315 OR 22467_8_321 OR 22467_8_322 OR 22467_8_324 OR 22467_8_332 OR 22467_8_343 OR 22467_8_345 OR 22467_8_348 OR 22467_8_349 OR 22467_8_355 OR 22467_8_356 OR 22467_8_368 OR 22467_8_375 OR 22467_7_94 OR 22467_8_194 OR 22467_8_195 OR 22467_8_197 OR 22467_8_200 OR 22467_8_202 OR 22467_8_206 OR 22467_8_208 OR 22467_8_211 OR 22467_8_218 OR 22467_8_222 OR 22467_8_224 OR 22467_8_228 OR 22467_8_241 OR 22467_8_243 OR 22467_8_246 OR 22467_7_44 OR 22467_7_45 OR 22467_7_46 OR 22467_7_47 OR 22467_7_52 OR 22467_7_60 OR 22467_7_64 OR 22467_7_67 OR 22467_7_68 OR 22467_7_69 OR 22467_7_75 OR 22467_7_77 OR 22467_7_78 OR 22467_7_79 OR 22467_7_82 OR 22467_7_86 OR 22467_7_87 OR 22467_7_89 OR 22467_7_91 OR 22467_7_317 OR 22467_7_318 OR 22467_7_319 OR 22467_7_320 OR 22467_7_331 OR 22467_7_336 OR 22467_7_337 OR 22467_7_343 OR 22467_7_35 OR 22467_7_358 OR 22467_7_36 OR 22467_7_360 OR 22467_7_361 OR 22467_7_37 OR 22467_7_373 OR 22467_7_374 OR 22467_7_376 OR 22467_8_252 OR 22467_8_253 OR 22467_8_254 OR 22467_8_257 OR 22467_8_258 OR 22467_8_259 OR 22467_8_262 OR 22467_8_270 OR 22467_8_271 OR 22467_8_288 OR 22467_8_292 OR 22467_8_294 OR 22467_8_306 OR 22467_8_308 OR 22467_7_214 OR 22467_7_216 OR 22467_7_218 OR 22467_7_220 OR 22467_7_222 OR 22467_7_225 OR 22467_7_226) AND facet_characteristic_name:(inferred_cell_type_-_authors_labels OR organism_part)
```

Among the results we have: `B cell` or `lymph node T cell`.

The same query for a set of lymph node cells:
```solr
cell_id:(21935_5_245 OR 21935_5_246 OR 21935_5_247 OR 21935_5_248 OR 21935_5_250 OR 21935_5_251 OR 21935_5_253 OR 21935_5_258 OR 21935_5_266 OR 21935_5_274 OR 21935_5_280 OR 21935_5_293 OR 21935_5_3 OR 21935_5_300 OR 21935_5_302 OR 21935_5_306 OR 21935_5_307 OR 21935_5_309 OR 21935_5_316 OR 21935_5_32 OR 21935_5_322 OR 21935_5_328 OR 21935_5_345 OR 21935_5_348 OR 21935_5_352 OR 21935_5_36 OR 21935_5_366 OR 21935_5_42 OR 21935_5_44 OR 21935_5_48 OR 21935_5_49 OR 21935_5_55 OR 21935_5_60 OR 21935_5_61 OR 21935_5_62 OR 21935_5_68 OR 21935_5_72 OR 21935_5_8 OR 21935_5_82 OR 21935_5_86 OR 21935_5_90 OR 21935_5_98 OR 21935_6_103 OR 21935_6_105 OR 21935_6_12 OR 21935_6_121 OR 21935_6_123 OR 21935_6_125 OR 21935_6_126 OR 21935_6_127 OR 21935_6_149 OR 21935_6_150 OR 21935_6_151 OR 21935_6_154 OR 21935_6_159 OR 21935_6_161 OR 21935_6_162 OR 21935_6_163 OR 21935_6_17 OR 21935_6_175 OR 21935_6_177 OR 21935_6_179 OR 21935_6_18 OR 21935_6_182 OR 21935_6_193 OR 21935_6_20 OR 21935_6_203 OR 21935_6_204 OR 21935_6_205 OR 21935_6_222 OR 21935_6_228 OR 21935_6_235 OR 21935_6_236 OR 21935_6_238 OR 21935_6_239 OR 21935_6_240 OR 21935_6_245 OR 21935_6_249 OR 21935_7_332 OR 21935_7_34 OR 21935_7_342 OR 21935_7_350 OR 21935_7_40 OR 21935_7_42 OR 21935_7_54 OR 21935_7_56 OR 21935_7_60 OR 21935_7_64 OR 21935_7_90 OR 21935_8_100 OR 21935_8_106 OR 22011_1_167 OR 22011_1_17 OR 22028_2_109 OR 22028_2_115 OR 22028_2_119 OR 22028_2_121 OR 22028_2_128 OR 22028_2_134 OR 22028_2_140) AND facet_characteristic_name:(inferred_cell_type_-_authors_labels OR organism_part)